commander3/src/comm_tod_mod.f90

```diff
@@ -569,46 +710,55 @@ subroutine read_hdf_scan(self, tod, filename, scan, ndet, detlabels)
     ! Read common scan data
     call read_hdf(file, slabel // "/common/vsun",  self%v_sun)
     call read_hdf(file, slabel // "/common/time",  self%t0)
-    call read_hdf(file, slabel // "/common/satpos",  self%satpos)
+    call read_hdf(file, slabel // "/common/satpos",  self%satpos, opt=.true.)
     call wall_time(t2)
     t_tot(1) = t2-t1
 
+    !write(*,*) self%t0(1), real(self%v_sun), "# v"
+
     ! Read detector scans
     allocate(self%d(ndet), buffer_sp(n))
     do i = 1, ndet
+       allocate(self%d(i)%psi(nhorn), self%d(i)%pix(nhorn))
+
        call wall_time(t1)
        field = detlabels(i)
        call wall_time(t2)
        t_tot(2) = t_tot(2) + t2-t1
        call wall_time(t1)
-       allocate(self%d(i)%tod(m))
        self%d(i)%label = trim(field)
        call read_hdf(file, slabel // "/" // trim(field) // "/scalars",   scalars)
        self%d(i)%gain_def   = scalars(1)
-       self%d(i)%sigma0_def = scalars(2) * self%d(i)%gain_def  ! To get sigma0 in uncalibrated units
+       self%d(i)%sigma0_def = scalars(2) * abs(self%d(i)%gain_def)  ! To get sigma0 in uncalibrated units
        self%d(i)%fknee_def  = scalars(3)
        self%d(i)%alpha_def  = scalars(4)
        self%d(i)%gain       = self%d(i)%gain_def
        self%d(i)%sigma0     = self%d(i)%sigma0_def
        self%d(i)%fknee      = self%d(i)%fknee_def
        self%d(i)%alpha      = self%d(i)%alpha_def
+
        call wall_time(t2)
        t_tot(3) = t_tot(3) + t2-t1
        call wall_time(t1)
-       call read_hdf(file, slabel // "/" // trim(field) // "/tod",    buffer_sp)
-       if (tod%halfring_split == 2 )then
-         self%d(i)%tod = buffer_sp(m+1:2*m)
-       else
-         self%d(i)%tod = buffer_sp(1:m)
-       end if
+       !call read_hdf(file, slabel // "/" // trim(field) // "/tod",    buffer_sp)
```

Comment:
This will break the halfring split functionality I think. I don't think we should merge this. Does it break WMAP or was it just commented out because no one knew what it was?

Reply:
the second one

Reply:
Hmm... I don't think anybody is working with halfrings right now, so my preference is to merge it now, and fix it later. I didn't figure out how to drop a single change in the GitHub interface..